### PR TITLE
[core] Use auto DH params with openssl3

### PR DESCRIPTION
**Pull request by xadhoom**
This fixes https://github.com/signalwire/freeswitch/issues/2195

From OpenSSL 3 docs for [SSL_CTX_set0_tmp_dh_pkey](https://www.openssl.org/docs/man3.0/man3/SSL_CTX_set0_tmp_dh_pkey.html):

`Applications may supply their own DH parameters instead of using the built-in values. This approach is discouraged and applications should in preference use the built-in parameter support described above. Applications wishing to supply their own DH parameters should call SSL_CTX_set0_tmp_dh_pkey() or SSL_set0_tmp_dh_pkey() to supply the parameters for the SSL_CTX or SSL respectively. The parameters should be supplied in the dhpkey argument as an EVP_PKEY containing DH parameters. Ownership of the dhpkey value is passed to the SSL_CTX or SSL object as a result of this call, and so the caller should not free it if the function call is successful.`

So the code in https://github.com/signalwire/freeswitch/blob/master/src/switch_rtp.c#L3738 is partially wrong because if the call of `SSL_CTX_set0_tmp_dh_pkey` is successful the key should *not* be freed, so `EVP_PKEY_free(dh_pk)` should not be called. But this is also discouraged, and using `SSL_set_dh_auto` is preferred.

This PR just does that, which result the segfault reported in #2195 to not happen anymore, because basically FS was freeing a key that was also freed by the underlying SSL_CTX_free when tearing down dtls.

### CODE REVIEW

File: src/switch_rtp.c [hunk 1]
```diff
@@ -3726,21 +3724,19 @@ SWITCH_DECLARE(switch_status_t) switch_rtp_add_dtls(switch_rtp_t *rtp_session, d
...
 		SSL_CTX_set_tmp_dh(dtls->ssl_ctx, dh);
 		DH_free(dh);
 	}
-#else 
-	if((dh_pk = PEM_read_bio_PrivateKey(bio, NULL, NULL, NULL)) != NULL) {
-		SSL_CTX_set0_tmp_dh_pkey(dtls->ssl_ctx, dh_pk);
-		EVP_PKEY_free(dh_pk);
-	}
-
 	BIO_free(bio);
```
Comment (andywolk) [inline src/switch_rtp.c:3735]: Double free?

**Review by andywolk — CHANGES_REQUESTED** [review]

File: src/switch_rtp.c (hunk 1 above)
Comment (xadhoom) [reply src/switch_rtp.c:3735]: Yes, explicit here and implicit by SSL_CTX_free, as per docs ```Ownership of the dhpkey value is passed to the SSL_CTX or SSL object as a result of this call, and so the caller should not free it if the function call is successful.```

Comment (xadhoom) [reply src/switch_rtp.c:3735]: Ok, got it, my error on reporting the patch. will fix immediately.

Comment (xadhoom) [reply src/switch_rtp.c]: fixed

**Review by andywolk — APPROVED** [review]

**Pull request merged**